internal/pkg/workspace/workspace.go

```diff
@@ -72,65 +72,75 @@ type Workspace struct {
 	logger        func(format string, args ...interface{})
 }
 
-// New returns a workspace, used for reading and writing to user's local workspace.
-func New() (*Workspace, error) {
-	fs := afero.NewOsFs()
-	fsUtils := &afero.Afero{Fs: fs}
-	logger := log.Infof
-
-	workingDirAbs, err := os.Getwd()
+// Use returns an existing workspace, searching for a copilot/ directory from the current wd,
+// up to 5 levels above. It returns ErrWorkspaceNotFound if no copilot/ directory is found.
+func Use(fs *afero.Afero, workingDirAbs string) (*Workspace, error) {
```

Comment:
I didn't read the PR 😂 I thought I could leave this early feedback to see what you think:
1. Is it possible to use the [`afero.Fs` interface](https://pkg.go.dev/github.com/spf13/afero#Fs) as the input instead of the concrete type? or are we depending on some more specific functionality

2. Can we make the "workingDirAbs" optional? could the constructor by default use `os.Getwd` unless the user specified `workspace.Use(&afero.Afero{Fs: afero.NewOsFs()}, workspace.WithWorkingDir("some/other/path"))`
Would that simplify the life of the `cli` pkg?

Reply:
Addressed both!

The whole reason why we needed to pass `workingDir` as a param was for the unit tests - specifically, the ones that test that Copilot is able to find a workspace in a parent directory.  I fixed the unit tests such that it doesn't need to do so. This way, we don't need it as a param anymore, not even optional.